pyne/alara.py

```diff
@@ -774,3 +806,43 @@ def _output_flux(ve, tag_flux,output,start,stop,direction):
 
     output += "\n\n"
     return output
+
+def _get_subvoxel_array(mesh, cell_mats):
+    """
+    This function returns an array of subvoxels.
+    Parameters
+    ----------
+    mesh : PyNE Mesh object
+        The Mesh object for which the geometry is discretized.
+
+    return : subvoxel_array: structured array
+        A sorted, one dimensional array, each entry containing the following
+        fields:
+
+            :svid: int
+                The index of non-void subvoxel id
+            :idx: int
+                The idx of the voxel
+            :scid: int
+                The cell index of the cell in that voxel
+
+    """
+    cell_number_tag = mesh.mesh.getTagHandle('cell_number')
+    subvoxel_array = np.zeros(0, dtype=[(b'svid', np.int64),
```

Comment:
What does `shape=0` mean here?  Does it just set up the structure but create an array with no entries?

Reply:
Yes, this line just create an array frame without entries. Because how many entries are there before the loop, so I create a frame and fill in entries in the loop.